contrib/completion/fish/docker.fish

```diff
@@ -23,6 +23,20 @@ function __fish_docker_no_subcommand --description 'Test if docker has yet to be
     return 0
 end
 
+function __fish_docker_seen_subcommands --description 'Detect a subcommand'
+    set -l cmd (commandline -poc)
+    set -e cmd[1]
+    if not [ (count $cmd) -eq (count $argv) ]
+        return 1
+    end
```

Comment:
I agree with all your other commits, but this one has the disadvantage that it is too strict.

A commandline like `docker -D run` will no longer get completions.

With [`argparse`](https://fishshell.com/docs/current/cmds/argparse.html) it's possible to get the best of both worlds.
Simply pass `$cmd` to an `argparse` command that knows Docker's global options.

```fish
$ set cmd --log-level info --debug run ubuntu
$ set -e cmd[1]
$ argparse -s D/debug=? l/log-level= -- $cmd; echo $argv
run ubuntu
```

and then compare `$argv[1]` (and more if you need)